File: docs/examples/context/test_example.py
from pathlib import Path

import pytest

from oso import Oso
import imp
import os


@pytest.fixture
def oso():
    return Oso()


@pytest.fixture
def load(oso):
    def load(policy):
        oso.load_file(Path(__file__).parent / policy)

    return load


def test_policy(oso, load):
    load("01-context.polar")
    imp.load_source("context", "02-context.py")

    oso._load_queued_files()

    os.environ["ENV"] = "production"
    assert not oso.allow("steve", "test", "policy")
    os.environ["ENV"] = "development"
    assert oso.allow("steve", "test", "policy")

File: docs/examples/context/java/TestExample.java
import java.io.IOException;

public class TestExample {
    public static void main(String[] args) {
        try {
            Oso oso = new Oso();
            oso.registerClass(Env.class, m -> new Env(), "Env");
            oso.loadFile("../01-context.polar");
            if (oso.allow("steve", "test", "policy")) {
                System.out.println("Works");
            } else {
                System.out.println("Doesn't work");
            }
        } catch (Exceptions.DuplicateClassAliasError e) {
            System.out.println(e);
        } catch (Exceptions.OsoException e) {
            System.out.println(e);
        } catch (IOException e) {
            System.out.println(e);
        }
    }
}